src/main/js/controls/Bubble/BubbleSingleDataset.js

```diff
@@ -0,0 +1,200 @@
+import { reflowLegend } from "../../helpers/legend";
+import {
+    createRegion,
+    hideAllRegions,
+    translateRegion,
+    areRegionsIdentical
+} from "../../helpers/region";
+import styles from "../../helpers/styles";
+import utils from "../../helpers/utils";
+import {
+    clear,
+    clickHandler,
+    hoverHandler,
+    prepareLegendItems,
+    processDataPoints,
+    translateBubbleGraph,
+    getDataPointValues,
+    calculateValuesRange
+} from "./helpers/helpers";
+import { draw, drawBubbles } from "./helpers/helpersSingleDataset";
+import Bubble from "./Bubble";
+
+/**
+ * A Bubble graph is a graph used to represent a collection of data
+ * points connected by a Bubble along the X and Y Axis.
+ *
+ * Lifecycle functions include:
+ *  * Load
+ *  * Generate
+ *  * Unload
+ *  * Destroy
+ *
+ * @module BubbleSingleDataset
+ * @class BubbleSingleDataset
+ */
+class BubbleSingleDataset extends Bubble {
+    /**
+     * @inheritdoc
+     * @public
+     */
+    load(graph) {
+        if (graph.content.length > 1) {
+            console.warn("BubbleSingleDataset can only use one dataset.");
+            return this;
+        }
+
+        if (
+            utils.isUndefined(graph.content[0].config.color) &&
+            utils.isDefined(graph.content[0].config.palette) &&
+            graph.content[0].config.values.length >
+                BUBBLE.MAX_BUBBLES_SINGLE_DATASET
+        ) {
+            console.warn("Cannot plot more than 4 bubbles using shades.");
+            return this;
+        }
+
+        this.dataTarget = processDataPoints(graph.config, this.config);
+        draw(graph.scale, graph.config, graph.svg, this.dataTarget);
+
+        if (utils.notEmpty(this.dataTarget.regions)) {
+            createRegion(
+                graph.scale,
+                graph.config,
+                graph.svg.select(`.${styles.regionGroup}`),
+                this.dataTarget.regions,
+                `region_${this.dataTarget.key}`,
+                this.config.yAxis
+            );
+        }
+
+        prepareLegendItems(
+            graph.config,
+            {
+                clickHandler: clickHandler(
+                    graph,
+                    this,
+                    graph.config,
+                    graph.svg
+                ),
+                hoverHandler: hoverHandler(graph.config.shownTargets, graph.svg)
+            },
+            this.dataTarget,
+            graph.legendSVG
+        );
+
+        prepareLabelShapeItem(
+            graph.config,
+            this.dataTarget,
+            graph.axesLabelShapeGroup[this.config.yAxis]
+        );
+
+        return this;
+    }
+
+    // /**
+    //  * @inheritdoc
+    //  */
+    // unload(graph) { ... }
+    //  inherited from class Bubble
+
+    /**
+     * @inheritdoc
+     */
+    resize(graph) {
+        if (utils.notEmpty(this.dataTarget.regions)) {
+            if (graph.content.length > 1 && !graph.config.shouldHideAllRegion) {
```

Comment:
would we ever fall into this true condition since a BubbleSingleDataset can only have one dataset?

Reply:
I tested it and seems like we don't. I removed this block here:
https://github.com/cerner/carbon-graphs/pull/308/commits/c3397298a95b82dd7a13c5bbe2e76e166cb29c83